xml-conduit/Text/XML/Stream/Parse.hs

```diff
@@ -567,25 +562,67 @@ parseIdent =
     valid '"'  = False
     valid '\'' = False
     valid '/'  = False
+    valid ';'  = False
+    valid '#'  = False
```

Comment:
This isn't a regression, but now that I've checked, XML specification actually:
- forbids characters that are allowed by this function
- allows characters that are forbidden by this function (notably `:`)

```
EntityRef ::= '&' Name ';'
NameStartChar ::= ":"  | [A-Z] | "_" | [a-z] | [#xC0-#xD6] | [#xD8-#xF6] | [#xF8-#x2FF] | [#x370-#x37D] | [#x37F-#x1FFF] | [#x200C-#x200D] | [#x2070-#x218F] |  [#x2C00-#x2FEF] | [#x3001-#xD7FF] | [#xF900-#xFDCF] | [#xFDF0-#xFFFD] | [#x10000-#xEFFFF]
NameChar ::= NameStartChar | "-" | "." | [0-9] | #xB7 | [#x0300-#x036F] | [#x203F-#x2040]
Name ::= NameStartChar (NameChar)*
```